jina/helper.py

```diff
@@ -107,7 +107,10 @@ def batch_iterator(data: Iterable[Any], batch_size: int, axis: int = 0,
         data = iter(data)
         # as iterator, there is no way to know the length of it
         while True:
-            chunk = tuple(islice(data, batch_size))
+            if yield_dict:
```

Comment:
better to add before `elif(data, Iterable)` an `elif(data, dict)`, like this it will work without the need of the `yield_dict` parameter.

Reply:
`data` needs to be sliced before passing it to `batch_iterator` because we have `num_batch` that inturn sets the  `total_size` . There are two options i think : 
1. Pass `total_size` to `batch_iterator` and slice the data inside the `batch_iterator`
2. Slice it before passing it to `batch_iterator` and pass 'yield_dict`. In this case dict when sliced is converted to a iterable .
wdyt?

Reply:
OK I see, let's keep passing `yield_dict` as argument so that we do not affect current behavior (the code is a little verbose already)